lib/std/os/windows.zig

```diff
@@ -172,6 +172,17 @@ pub fn GetCurrentThreadId() DWORD {
     return @truncate(@intFromPtr(teb().ClientId.UniqueThread));
 }
 
+pub fn GetLastError() Win32Error {
+    const offset = comptime switch (builtin.target.ptrBitWidth()) {
+        32 => 0x34,
+        64 => 0x68,
+        else => @compileError("Unkown target."),
+    };
```

Comment:
Would be better to add a field to the `TEB` struct and then access that. I believe this offset might correspond to the first pointer in Zig's current `TEB.Reserved2` field, but definitely double-check that.

Reply:
The proposed change would break the TEB in 32-bit Windows, because of the difference in PVOID size. We could do this:

```diff
-     Reserved2: [399]PVOID,
+     LastErrorValue: ULONG,
+     Reserved2: [399 * @sizeOf(PVOID) - @sizeOf(ULONG)]u8,
```

Reply:
That works, good catch. Can also add offset checks here:

https://github.com/ziglang/zig/blob/ee6a52b40f9836beb0d35820d1987f3c7e522f45/lib/std/os/windows.zig#L4443-L4460